build.gradle.kts

```diff
@@ -25,6 +25,32 @@ val mokoVersion = libs.versions.mokoPermissionsVersion.get()
 allprojects {
     group = "dev.icerock.moko"
     version = mokoVersion
+    
+    // Configure Java toolchain to use Java 17
+    plugins.withId("org.jetbrains.kotlin.multiplatform") {
+        extensions.configure<org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension> {
+            jvmToolchain(17)
+        }
+    }
+    
+    // For Android projects
+    plugins.withId("com.android.application") {
+        extensions.configure<com.android.build.gradle.BaseExtension> {
+            compileOptions {
+                sourceCompatibility = JavaVersion.VERSION_17
+                targetCompatibility = JavaVersion.VERSION_17
+            }
+        }
+    }
+    
+    plugins.withId("com.android.library") {
+        extensions.configure<com.android.build.gradle.LibraryExtension> {
+            compileOptions {
+                sourceCompatibility = JavaVersion.VERSION_17
+                targetCompatibility = JavaVersion.VERSION_17
+            }
+        }
+    }
```

Comment:
instead of this change we need just update moko-gradle-plugin dependency. 

https://github.com/icerockdev/moko-permissions/blob/d6aded96b30aebe64bba6730319800400ab50e29/gradle/libs.versions.toml#L37

this line should be changed to latest release of https://github.com/icerockdev/moko-gradle-plugin/releases